### DISCUSSION TITLE
question: How can I convert mysql errors to `HandlerError` ?

### DISCUSSION THREAD

**Discussion by antonio-hickey** [aws/aws-lambda-rust-runtime#550] [Q&A] [answered]
Hey all, sorry in advance for putting a question in issues, but I'm having trouble converting error's from [mysql crate](https://docs.rs/mysql/latest/mysql/) to `HandlerError` was wondering if anyone could assist.

The mysql crate docs has this:
```
let pool = Pool::new(url)?;
```

I tried this with no luck:
```
let pool = Pool::new(url).map_err(|e| HandlerError::from(format!("{:?}", e).as_str()))?;
```

Any help is appreciated thanks!

**Comment by calavera** [comment, marked as answer]
You can return in your function anything that implements `fmt::Display + fmt::Debug`. `HandlerError` doesn't exist anymore in the stable version of the runtime. If you're using an old version of the runtime, I'd recommend you to update to the latest stable version.

**Reply by antonio-hickey** [reply]
Yup this helped a lot, not sure why I didn't think to check version 🤦‍♂️ was on version 0.1, so upgraded to 0.7

**Reply by calavera** [reply]
I have not had time to update the article that you probably followed, sorry 😭

**Reply by antonio-hickey** [reply]
This was the article if it helps, https://aws.amazon.com/blogs/opensource/rust-runtime-for-aws-lambda/

**Reply by calavera** [reply]
yup, you're not the only one following those instructions, sorry about that.

**Reply by DontBreakAlex** [reply]
Can you elaborate on this answer ? My function returns this:
```Rust
#[derive(Error, Debug)]
pub enum GenApiKeyError {
	#[error("Could not find CSRF token in login page")]
	CsrfError,
	#[error("Login failed")]
	LoginError,
	#[error("Request failed: {0}")]
	RequestError(#[from] reqwest::Error),
	#[error("Failed to parse body: {0}")]
	ParseError(#[from] Utf8Error),
	#[error("Failed get config: {0}")]
	ConfigError(#[from] envconfig::Error),
}
```
`thiserror` implements `Display`, but I still get ```{"type":"https://httpstatuses.com/500","status":500,"title":"Internal Server Error","detail":"failed to serialize lambda-url event"}```
